Code/IO/src/sitkImageReaderBase.cxx

```diff
@@ -78,8 +81,26 @@ itk::SmartPointer<ImageIOBase>
 ImageReaderBase
 ::GetImageIOBase(const std::string &fileName)
 {
-  itk::ImageIOBase::Pointer iobase =
-    itk::ImageIOFactory::CreateImageIO( fileName.c_str(), itk::ImageIOFactory::ReadMode);
+  itk::ImageIOBase::Pointer iobase;
+  if (this->m_ImageIOName == "")
+    {
+    iobase = itk::ImageIOFactory::CreateImageIO( fileName.c_str(), itk::ImageIOFactory::ReadMode);
+    }
```

Comment:
Should this setting also have a side effect of updating the m_ImageIOName using the GetNameIOfClass method, or should it leave the IO name as the empty string so that reading another file with different extension will work too. I think the later (current behavior) is expected, but not sure.